### PR TITLE
KHWEBA-19. Add tags

### PR DESCRIPTION
**Pull request by Vladyslav-Plakhuta**
created new KHWEBA-19 branch and pushed all files which were updated previously

### CODE REVIEW

File: src/components/add-tag-dialog/add-tag-dialog.js [hunk 1]
```diff
@@ -34,6 +31,21 @@ export class AddTagComponent {
...
+
+  handleOk() {
+    this.props.onAddTagConfirm({
+      name: this.tagNameInput.value
+    });
+    this.clean();
+  }
+
+  handleCancel() {
+    console.log("declined");
```
Comment (gift-a) [inline src/components/add-tag-dialog/add-tag-dialog.js]: you need this handle to clean your dialog. it is no need in `console.log` anymore

File: src/components/drawer/drawer.js [hunk 2]
```diff
@@ -12,29 +13,43 @@ export class DrawerComponent {
...
     this.moreBtnMountPoints = this.mountPoint.querySelectorAll(
       ".drawer__more-button"
     );
   }
+
   updateAccountData(transaction) {
     this.accountsComponent.updateAccountData(transaction);
   }
 
+  updateTagData(transaction) {
```
Comment (gift-a) [inline src/components/drawer/drawer.js]: what does this method do? do you have this method in tags component?

File: src/components/drawer/drawer.js [hunk 3]
```diff
@@ -12,29 +13,43 @@ export class DrawerComponent {
...
+    this.tagsComponent.updateTagData(transaction);
+  }
+
   updateAccountDataDelete(transaction) {
     this.accountsComponent.updateAccountDataDelete(transaction);
   }
 
+  updateTagDataDelete(transaction) {
```
Comment (gift-a) [inline src/components/drawer/drawer.js]: what does this method do? do you have this method in tags component?

File: src/components/tags/tags.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
+import template from "./tags.html";
+import tagItemTemplate from "./tag-item.html";
+import { ButtonMoreComponent } from "../button-more/button-more";
+import { TagListService } from "../../services/tag-service";
+import { AddTagDialogComponent } from "../add-tag-dialog/add-tag-dialog";
+
+export class TagsComponent {
+  constructor(mountPoint, props) {
```
Comment (gift-a) [inline src/components/tags/tags.js]: you don't have props when you init your component

File: src/components/tags/tags.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+  handleDeleteClick() {
+    console.log("handleDeleteClick");
+  }
+
+  mount() {
+    this.mountPoint.innerHTML = template();
+    this.querySelectors();
+    this.initMoreBtns();
+    this.addEventListeners();
+    this.initAddTagDialogComponent();
```
Comment (gift-a) [inline src/components/tags/tags.js]: it should name `mountChildren`

**Comment by Vladyslav-Plakhuta** [comment]
the conflict has been resolved

review pls

File: src/components/tags/tags.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+  }
+
+  handleDeleteClick() {
+    console.log("handleDeleteClick");
+  }
+
+  mount() {
+    this.mountPoint.innerHTML = template();
+    this.querySelectors();
+    this.initMoreBtns();
```
Comment (gift-a) [inline src/components/tags/tags.js]: it seems you don't have buttons mount points before you load stored tags, so you don't need `this.initMoreBtns();` and `this.moreBtnMountPoints` on the row 15
try it. I think it will work without it

File: src/components/tags/tags.js [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+        TagListService.set(tags);
+      }
+      this.initTags(tags);
+    });
+  }
+
+  updateTagData(transaction) {
+    TagListService.update(
+      transaction.tag,
+      parseInt(transaction.type + transaction.amount)
```
Comment (gift-a) [inline src/components/tags/tags.js]: what does `parseInt` do here? do you have numbers?

File: src/components/add-tag-dialog/add-tag-dialog.js [hunk 8]
```diff
@@ -1,37 +1,44 @@
...
 
   showDialog() {
     this.dialog.show();
   }
 
-  handleOk() {
-    console.log("accepted");
-  }
-
-  handleCancel() {
```
Comment (gift-a) [inline src/components/add-tag-dialog/add-tag-dialog.js]: why did you delete this method? you should clean your dialog with this method

File: src/components/drawer/drawer.js [hunk 9]
```diff
@@ -75,7 +72,7 @@ export class DrawerComponent {
     this.querySelectors();
     this.initMoreBtns();
```
Comment (gift-a) [inline src/components/drawer/drawer.js]: you don't need more buttons in drawer anymore. they are in tags and amounts components

File: src/components/tags/tags.js [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+import { AddTagDialogComponent } from "../add-tag-dialog/add-tag-dialog";
+
+export class TagsComponent {
+  constructor(mountPoint) {
+    this.mountPoint = mountPoint;
+  }
+
+  querySelectors() {
+    this.tagsRoot = this.mountPoint.querySelector(".tags");
+    this.moreBtnMountPoints = this.mountPoint.querySelectorAll(
```
Comment (gift-a) [inline src/components/tags/tags.js]: you don't need this, because you firstly mount empty component (no buttons to search), then you load tags from storage and get your buttons

File: src/components/tags/tags.js (hunk 5 above)
Comment (gift-a) [reply src/components/tags/tags.js]: it is naming conversation. in future you will have confirm dialog. and instead of initialisating confirm dialog you will place it in `mountChildren`

File: src/components/drawer/drawer.js (hunk 9 above)
Comment (Vladyslav-Plakhuta) [reply src/components/drawer/drawer.js]: I have not made changes in drawer in part of this.initMoreBtns();

Comment (ihor7016) [reply src/components/drawer/drawer.js]: yes but you do not need this method anymore -- so delete it

File: src/components/tags/tags.js [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+    this.addTagButton = this.mountPoint.querySelector(
+      ".tags__add-tag-dialog-activation"
+    );
+    this.addTagMountPoint = document.querySelector(".tags__add-tag-dialog");
+    this.tagsList = this.mountPoint.querySelector(".tags__list-items");
+  }
+
+  handleAddTagConfirmed(tag) {
+    TagListService.add(tag);
+    this.addTagToHead(tag);
```
Comment (ihor7016) [inline src/components/tags/tags.js:25]: you do not init 3dot button after adding tag

image

File: src/components/tags/tags.scss [hunk 12]
```diff
@@ -0,0 +1,14 @@
+.tags {
+  &__more-button {
```
Comment (ihor7016) [inline src/components/tags/tags.scss]: please extract it into reusable BEM (scss only) component and use here and in accounts component

**Review by gift-a — APPROVED** [review]

**Review by ihor7016 — APPROVED** [review]
